azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -922,12 +933,23 @@ func flattenAzureRmVirtualMachineScaleSetStorageProfileDataDisk(disks *[]compute
 	return result
 }
 
-func flattenAzureRmVirtualMachineScaleSetStorageProfileImageReference(profile *compute.ImageReference) []interface{} {
+func flattenAzureRmVirtualMachineScaleSetStorageProfileImageReference(image *compute.ImageReference) []interface{} {
 	result := make(map[string]interface{})
-	result["publisher"] = *profile.Publisher
-	result["offer"] = *profile.Offer
-	result["sku"] = *profile.Sku
-	result["version"] = *profile.Version
+	if image.Publisher != nil {
+		result["publisher"] = *image.Publisher
+	}
+	if image.Offer != nil {
+		result["offer"] = *image.Offer
+	}
+	if image.Sku != nil {
+		result["sku"] = *image.Sku
+	}
+	if image.Version != nil {
+		result["version"] = *image.Version
+	}
+	if image.ID != nil {
+		result["id"] = *image.ID
+	}
```

Comment:
I have a feeling it shouldn't - but does this change the hash? i.e. when having previously built a resource using the VMSS resource in `master`, then running `terraform plan` against this branch - are there any changes shown?